report/latest_map.go

```diff
@@ -193,15 +193,57 @@ func (m *LatestMap) CodecEncodeSelf(encoder *codec.Encoder) {
 	} else {
 		encoder.Encode(nil)
 	}
+
+	//_, r := codec.GenHelperEncoder(encoder)
+	//if m.Map == nil {
+	//	r.EncodeNil()
+	//	return
+	//}
+	//
+	//r.EncodeMapStart(m.Map.Size())
+	//m.Map.ForEach(func(key string, value interface{}) {
+	//	r.EncodeString(1, key)
+	//	encoder.Encode(value.(LatestEntry))
+	//})
 }
 
+const (
+	containerMapKey   = 2
+	containerMapValue = 3
+	containerMapEnd   = 4
+)
+
 // CodecDecodeSelf implements codec.Selfer
 func (m *LatestMap) CodecDecodeSelf(decoder *codec.Decoder) {
```

Comment:
Please add a comment explaining that the complexity in this function is because the naive solution is much slower, maybe even referring to this PR.

Otherwise, someone unfamiliar with recent events might come along, look at this, see that it's really complicated & could be replaced with a simple call to `Decode` & `fromIntermediate`.